src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = item[key] as string;
+
+    if (!result[value]) {
+      result[value] = [];
+    }
+
+    result[value].push(item);
+  });
+
+  return result;
```

Comment:
See how you're using an external object in your `.forEach`?  When you want to transform an array into one single value (in this case an object) prefer `.reduce` which eliminates the usage of an external variable